factorial_recursive.py

```diff
@@ -0,0 +1,11 @@
+def factorial(n):
```

Comment:
Please copy the type hints and doctests from https://github.com/TheAlgorithms/Python/blob/master/maths/factorial_python.py#L1-L17 and paste them here and ensure that they run and pass locally.